src/main/java/roomescape/repository/ReservationRepository.java

```diff
@@ -0,0 +1,61 @@
+package roomescape.repository;
+
+import java.util.List;
+import javax.sql.DataSource;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+
+@Repository
+public class ReservationRepository {
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert jdbcInsert;
+
+    public ReservationRepository(final JdbcTemplate jdbcTemplate, final DataSource dataSource) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.jdbcInsert = new SimpleJdbcInsert(dataSource)
+                .withTableName("reservation")
+                .usingGeneratedKeyColumns("id");
+    }
+
+    public Long save(final Reservation reservation) {
+        SqlParameterSource parameters = new MapSqlParameterSource()
+                .addValue("name", reservation.getName())
+                .addValue("date", reservation.getDate())
+                .addValue("time_id", reservation.getTimeId());
+        return jdbcInsert.executeAndReturnKey(parameters).longValue();
+    }
```

Comment:
```suggestion
    public Long save(final Reservation reservation) {
        SqlParameterSource parameters = new MapSqlParameterSource()
                .addValue("name", reservation.getName())
                .addValue("date", reservation.getDate())
                .addValue("time_id", reservation.getTimeId());
        long id = jdbcInsert.executeAndReturnKey(parameters).longValue();
        return new Reservation(id, reservation.getName(), reservation.getDate(), reservation.getTimeId());
    }
```
저장후 id만 반환하기보다는 저장된 도메인을 반환하는게 어떨까요?

Reply:
전체적인 흐름이 바뀔수있어서 참고만해주세요!

Reply:
리팩토링을 진행하면서 이 부분을 다시 한번 고민해봤습니다.

만약 현재 로직에서 `id`를 반환하는 것이 아니라, 도메인을 반환하게 되면 `Reservation`의 필드인 `ReservationTime` 객체도 `timeId`로 찾아와야 하는 문제가 있습니다. 즉, `save` 메소드에서 저장과 조회 두 가지 일을 하게 될 가능성이 있어서 저는 `id`를 반환하고 Service 계층에서 `id`를 활용해 여러 비즈니스 로직을 수행하는 편이 더 낫다고 생각했습니다.

Reply:
lazy loading에 대해서도 나중에 쉽게 이해하실 수 있겠네요!